app/services/search/query_builders/classified_listing.rb

```diff
@@ -0,0 +1,124 @@
+module Search
+  module QueryBuilders
+    class ClassifiedListing
+      TERM_KEYS = %i[
+        category
+        contact_via_connect
+        location
+        slug
+        tags
+        title
+        user_id
+      ].freeze
+
+      RANGE_KEYS = %i[
+        bumped_at
+        expires_at
+      ].freeze
+
+      QUERY_KEYS = %i[
+        classified_listing_search
+      ].freeze
+
+      DEFAULT_PARAMS = {
+        sort_by: "bumped_at",
+        sort_direction: "desc",
+        size: 0
+      }.freeze
+
+      attr_accessor :params, :body
+
+      def initialize(params)
+        @params = params.deep_symbolize_keys
+        build_body
+      end
+
+      def as_hash
+        @body
+      end
+
+      private
+
+      def build_body
+        @body = ActiveSupport::HashWithIndifferentAccess.new
+        build_queries
+        add_sort
+        set_size
+      end
+
+      def build_queries
+        @body[:query] = { bool: {} }
+        @body[:query][:bool][:filter] = filter_conditions if filter_keys_present?
+        @body[:query][:bool][:must] = query_conditions if query_keys_present?
+      end
+
+      def add_sort
+        sort_key = @params[:sort_by] || DEFAULT_PARAMS[:sort_by]
+        sort_direction = @params[:sort_direction] || DEFAULT_PARAMS[:sort_direction]
+        @body[:sort] = {
+          sort_key => sort_direction
+        }
+      end
+
+      def set_size
+        # By default we will return 0 documents if size is not specified
+        @body[:size] = @params[:size] || DEFAULT_PARAMS[:size]
+      end
+
+      def filter_conditions
+        filter_conditions = []
+        filter_conditions.concat term_keys if term_keys_present?
+        filter_conditions.concat range_keys if range_keys_present?
+
+        filter_conditions
+      end
+
+      def term_keys
+        TERM_KEYS.map do |term_key|
+          next if @params[term_key].blank? && @params[term_key] != false
```

Comment:
Suggestion, what if we used `next  unless @params.has_key?(term_key)` here? Would that work? Feels a little cleaner and it took me 5 min to walk through the true/false boolean logic

Reply:
Yeah, I actually hate this line. I got stuck in between `.present?`, `.empty?`, `.blank?`, etc. because we have to account for params having `false` as a value.

I'm down to just check the key, the only caveat is you can end up with a query that's like:
`term: "null"` because we aren't checking for the existence of the value with `.has_key?`. That would happen in a scenario where someone built params from variables not realizing the value of the variable was `nil`. But I don't think that's actually a big deal at all, Elasticsearch doesn't mind taking `"null"`.

I'm going to change it to `.has_key?` unless you have more thoughts about potentially passing `nil` to Elasticsearch?

Reply:
I think we can prevent that from happening through good code review and testing. If you do that the wrong results, if any, will be returned. But if you are concerned 
```
next unless @params.has_key?(term_key) && !@params[term_key].nil?
```

Reply:
That's true, I'm down with just the key check. That makes sense to me and it is more readable.

I believe `.key?` is generally preferred over `.has_key?` so I switched both the `term_keys` and `range_keys` check to have `return unless @params.key? term_key` and `return unless @params.key? range_key`